src/rust/ensogl/src/display/scene.rs

```diff
@@ -120,23 +129,84 @@ pub enum Target {
 }
 
 impl Target {
-    fn to_internal(&self) -> Vector4<u32> {
+
+    /// Encode two u32 values into three u8 values.
+    ///
+    /// This is the same encoding that is used in the `fragment_runner`. This encoding is lossy and
+    /// can only encode values up to 12^2=4096 each
+    ///
+    /// We use 12 bits from each value and pack them into the 3 output bytes like described in the
+    /// following diagram.
+    ///
+    ///
+    ///  Input
+    ///
+    ///    value1 (v1) as bytes               value2 (v2) as bytes
+    ///   +-----+-----+-----+-----+           +-----+-----+-----+-----+
+    ///   |     |     |     |     |           |     |     |     |     |
+    ///   +-----+-----+-----+-----+           +-----+-----+-----+-----+
+    /// 32    24    16     8      0         32    24    16     8      0   <- Bit index
+    ///
+    ///
+    /// Output
+    ///
+    /// byte1            byte2                     byte3
+    /// +-----------+    +-------------------+     +-----------+
+    /// | v1[12..4] |    | v1[4..0] v2[4..0] |     | v2[12..4] |
```

Comment:
Please use the notion of inclusive / non-incliusive ranges (I asked about it in my previous PR review). I believe it should be `v1[4=..=0]` and `v1[12=..4]` right? We can of course use mathematical notion instead, so `v1[12..4[` - according to your preferences.